common/djangoapps/adaptive_learning/config/models.py

```diff
@@ -3,15 +3,74 @@
 """
 
 from config_models.models import ConfigurationModel
+from openedx.core.djangoapps.xmodule_django.models import CourseKeyField
 
 
 class AdaptiveLearningEnabledFlag(ConfigurationModel):
     """
-    A flag that enables/disables Adaptive Learning across an entire instance.
+    A flag that enables/disables the Adaptive Learning feature
+    across an entire instance.
+
+    Defaults to False instance-wide.
     """
 
+    @classmethod
+    def feature_enabled(cls, course_id=None):
+        """
+        Looks at the currently active configuration model to determine
+        whether the Adaptive Learning is available (either for an instance
+        or for a course).
+
+        If the course_id parameter is passed, it returns a boolean determining
+        whether Adaptive Learning is available for the particular course.
+        Note that in order for Adaptive Learning to be available in a given course
+        both flags (an instance-wide and and a course-specific ones)
+        need to be enabled.
+
+        If the course_id parameter is not passed, it returns a boolean
+        specyfing whether Adaptive Learning feature is available instance-wide.
+        """
+
+        if AdaptiveLearningEnabledFlag.current().enabled:
+            if course_id:
+                course_flag = (CourseAdaptiveLearningFlag.objects
+                               .filter(course_id=course_id)
+                               .order_by('-change_date')
+                               .first())
+                return course_flag.enabled if course_flag else False
+            else:
+                return True
+        else:
+            return False
+
     class Meta(object):
         app_label = "adaptive_learning"
 
     def __unicode__(self):
-        return u"AdaptiveLearningEnabledFlag: enabled {}".format(self.enabled)
+        current_model = AdaptiveLearningEnabledFlag.current()
+        return u"AdaptiveLearningFlag: enabled {}".format(
```

Comment:
@tomaszgy Typo: "AdaptiveLearningFlag" should be "AdaptiveLearningEnabledFlag".

Reply:
👍